src/main/java/edu/cornell/kfs/pdp/businessobject/lookup/CuPayeeACHAccountLookupableHelperServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+import org.kuali.rice.core.api.criteria.PredicateUtils;
+import org.kuali.rice.core.api.criteria.QueryByCriteria;
+import org.kuali.rice.core.api.encryption.EncryptionService;
+import org.kuali.rice.kim.api.identity.Person;
+import org.kuali.rice.kim.api.identity.PersonService;
+import org.kuali.rice.kim.impl.KIMPropertyConstants;
+import org.kuali.rice.kns.lookup.LookupUtils;
+import org.kuali.rice.krad.bo.BusinessObject;
+import org.kuali.rice.krad.util.BeanPropertyComparator;
+import org.kuali.rice.krad.util.KRADConstants;
+
+import edu.cornell.kfs.pdp.CUPdpPropertyConstants;
+
+/**
+ * Custom override of PayeeACHAccountLookupableHelperServiceImpl that performs special handling
+ * when the new principal name search field is given a value.
+ */
+@SuppressWarnings("deprecation")
+public class CuPayeeACHAccountLookupableHelperServiceImpl extends PayeeACHAccountLookupableHelperServiceImpl {
+    private static final long serialVersionUID = 9164050782113012442L;
+
+    private PersonService personService;
+    private CriteriaLookupService criteriaLookupService;
+
+    /**
+     * Overridden to perform custom searching when a principal name is specified on the search screen.
+     * 
+     * @see org.kuali.rice.kns.lookup.KualiLookupableHelperServiceImpl#getSearchResultsHelper(java.util.Map, boolean)
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    protected List<? extends BusinessObject> getSearchResultsHelper(Map<String,String> fieldValues, boolean unbounded) {
+        if (StringUtils.isNotBlank(fieldValues.get(CUPdpPropertyConstants.PAYEE_PRINCIPAL_NAME))) {
+            List<PayeeACHAccount> results = null;
+            
+            // Search for people with the given principal name(s), in a manner that respects lookup criteria Strings.
+            List<Person> people = personService.findPeople(Collections.singletonMap(
+                    KIMPropertyConstants.Principal.PRINCIPAL_NAME, fieldValues.get(CUPdpPropertyConstants.PAYEE_PRINCIPAL_NAME)));
+            if (!people.isEmpty()) {
+                // Get the users' entity IDs and employee IDs for searching.
+                List<String> entityIds = new ArrayList<String>();
+                List<String> employeeIds = new ArrayList<String>();
+                for (Person person : people) {
+                    entityIds.add(person.getEntityId());
+                    if (StringUtils.isNotBlank(person.getEmployeeId())) {
+                        employeeIds.add(person.getEmployeeId());
+                    }
+                }
+                
+                // Create a map without blank values and with all encrypted values decrypted, similar to the ancestor class's logic.
+                Map<String,String> finalFieldValues = new HashMap<String,String>();
+                for (Map.Entry<String,String> entry : fieldValues.entrySet()) {
+                    // Only add non-blank values.
+                    if (StringUtils.isBlank(entry.getValue())) {

```

Comment:
I could have structured this "if" block to not need all these "else if" or "else" parts, but I did it this way instead in order to avoid CheckStyle complaints about nested "if" depth.